hushline/static/js/directory.js

```diff
@@ -45,6 +52,20 @@ document.addEventListener('DOMContentLoaded', function () {
         return text.replace(regex, '<span class="search-highlight">$1</span>');
     }
 
+
+    function reportUser(username, bio) {
+
+        // Construct the message content with explicit line breaks
+        const messageContent = `Reported user: ${username}\n\nBio: ${bio}\n\nReason:`;
```

Comment:
@glenn-sorrentino isn't this what I have per your comment? Still getting a "User not found error"

I'm looking at our `routes.py` route and don't see anything to support this? Unless I'm missing something? (Sorry for the confusion!)


https://github.com/scidsg/hushline/pull/400#issuecomment-2221300342

Reply:
Or could it be that I don't have an admin user on my local db?

Reply:
lol created an`admin` account on my local db and now it works. Sorry 😂!

Reply:
What I mean is that when a user clicks "Report Account" it should take then to the admin account's submit message page and have the Message textarea pre-populated with info on the account being reported.

Reply:
This is the function in the directory.js file:

```
    function reportUser(username, bio) {

        // Construct the message content with explicit line breaks
        const messageContent = `Reported user: ${username}\n\nBio: ${bio}\n\nReason:`;

        // Encode the message content to ensure line breaks and other special characters are correctly handled
        const encodedMessage = encodeURIComponent(messageContent);
        
        // Redirect to the message submission form for the admin with the pre-filled content
        const submissionUrl = `/submit_message/admin?prefill=${encodedMessage}`;
        window.location.href = submissionUrl;
    }
```